test/suite/test_alter05.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python
+#
+# Public Domain 2014-present MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#
+# This is free and unencumbered software released into the public domain.
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import wiredtiger, wttest
+from wtscenario import make_scenarios
+
+# test_alter05.py
+#    Check the alter command even if the file is modified.
+class test_alter05(wttest.WiredTigerTestCase):
+    name = "alter05"
+
+    def verify_metadata(self, metastr):
+        c = self.session.open_cursor('metadata:', None, None)
+
+        # We must find a file type entry for this object and its value
+        # should contain the provided file meta string.
+        c.set_key('file:' + self.name)
+        self.assertNotEqual(c.search(), wiredtiger.WT_NOTFOUND)
+        value = c.get_value()
+        self.assertTrue(value.find(metastr) != -1)
+
+        c.close()
+
+    # Alter file to change the metadata and verify
+    def test_alter05(self):
+        uri = "file:" + self.name
+        entries = 100
+        create_params = 'key_format=i,value_format=i,'
+
+        self.session.create(uri, create_params + "log=(enabled=true)")
+
+        # Put some data in table.
+        c = self.session.open_cursor(uri, None)
+        for k in range(entries):
+            c[k+1] = 1
+        c.close()
+
+        # Verify the string in the metadata
+        self.verify_metadata('log=(enabled=true)')
+
+        # Alter app metadata and verify
```

Comment:
I believe we can't be sure if we tried taking a checkpoint because the eviction is not deterministic and we have observed this while trying to reproduce the bug. Maybe we need to be explicit about the points in the test where we are trying to exercise the ability of WT to perform checkpoint internally.

Reply:
@hbokhari I feel it is difficult to generate a scenario where checkpoint only splits the page and then every marks it as dirty after the checkpoint. The PR change intention is to retry the WT_SESSION::alter operation after an EBUSY, so I just generated a simple test to demonstrate it.

Reply:
@kommiharibabu makes sense.

Reply:
@sueloverso Yes, it fails in the current tree if the btree is dirty. I can add stats to confirm it.